### PR TITLE
Update to iTwinUI v2

### LINKED ISSUE #85: Update to iTwinUI v2
After `appui` moves to iTwinUI v2 need to update `components` package and test-app to use v2 as well.

### PR DESCRIPTION
**Pull request by saskliutas**
Updated to use iTwinUI v2.
Updated appui to latest dev version that uses iTwinUI v2
Adjusted components styles to work correctly with iTwinUI v2

Closes #85

### CODE REVIEW

File: change/@itwin-presentation-components-7ac57041-25a5-494d-ab65-1f374048fd21.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Updated `@itwinui-react` package to ^2.5.0",
```
Comment (grigasp) [inline change/@itwin-presentation-components-7ac57041-25a5-494d-ab65-1f374048fd21.json:3]:
```suggestion
  "comment": "Updated `@itwinui-react` package to `^2.5.0`",
```

File: packages/components/package.json [hunk 2]
```diff
@@ -56,35 +57,36 @@
...
+    "@itwin/components-react": "^4.0.0-dev.11",
     "@itwin/core-bentley": "^3.6.0 || ^4.0.0-dev.7",
     "@itwin/core-common": "^3.6.0 || ^4.0.0-dev.7",
     "@itwin/core-frontend": "^3.6.0 || ^4.0.0-dev.7",
-    "@itwin/core-react": "^3.6.0 || 4.0.0-dev.3",
-    "@itwin/imodel-components-react": "^3.6.0 || 4.0.0-dev.3",
+    "@itwin/core-react": "^4.0.0-dev.11",
+    "@itwin/imodel-components-react": "^4.0.0-dev.11",
     "@itwin/presentation-common": "^3.6.0 || ^4.0.0-dev.7",
     "@itwin/presentation-frontend": "^3.6.0 || ^4.0.0-dev.7",
```
Comment (grigasp) [inline packages/components/package.json:68]: is there a point to keep some peer deps at `dev.7` and not upgrade them?

File: packages/components/src/presentation-components/instance-filter-builder/MultiTagSelect.tsx [hunk 3]
```diff
@@ -101,7 +102,7 @@ function TagLabel({ children, ...props }: any) {
 }
 
 function TagRemove(props: any) {
-  return <components.MultiValueRemove {...props} innerProps={{ ...props.innerProps, className: "iui-button iui-borderless iui-small iui-tag-button" }}>
+  return <components.MultiValueRemove {...props} innerProps={{ ...props.innerProps, className: "iui-button iui-tag-button", ["data-iui-variant"]: "borderless", ["data-iui-size"]: "size" }}>
```
Comment (grigasp) [inline packages/components/src/presentation-components/instance-filter-builder/MultiTagSelect.tsx:105]: I might be wrong, but should the value in `["data-iui-size"]: "size"` be `"small"`?

File: packages/components/package.json (hunk 2 above)
Comment (saskliutas) [reply packages/components/package.json]: Thanks, somehow missed some while updating.

File: packages/components/src/presentation-components/instance-filter-builder/MultiTagSelect.tsx (hunk 3 above)
Comment (saskliutas) [reply packages/components/src/presentation-components/instance-filter-builder/MultiTagSelect.tsx]: Nice catch, thanks.

**Review by grigasp — APPROVED** [review]

**Pull request merged**